dagshub/data_engine/model/datasource.py

```diff
@@ -1391,6 +1443,21 @@ def _test_not_comparing_other_ds(other):
         if type(other) is Datasource:
             raise DatasetFieldComparisonError()
 
+    @staticmethod
+    def _convert_file_to_df(file_path: str):
+        # prepare dataframe for import_metadata
+        if file_path.lower().endswith(".csv"):
+            df = pd.read_csv(file_path)
+        elif file_path.lower().endswith(".parquet"):
+            df = pd.read_parquet(file_path)
+        elif file_path.lower().endswith(".zip"):
+            df = pd.read_csv(file_path, compression="zip")
+        elif file_path.lower().endswith(".gz"):
+            df = pd.read_csv(file_path, compression="gzip")
+        else:
+            raise RuntimeError(f"Unsupported file format: {file_path}")
```

Comment:
Let the user know what they can import + a whole path is more than just the format, so the error could be more precise.
```suggestion
        else:
            raise RuntimeError(f"File '{file_path}' needs to be a .csv/.parquet or a compressed .zip/.gz to be imported")
```

btw, I would also look into how the compression flags work, because I expect users throwing a 10-file archives at this function and it failing in pandas. Can you check what pandas throws when you do that?

Reply:
In such cases it would throw:
```
                    raise ValueError(
                        "Multiple files found in ZIP file. "
                        f"Only one file per ZIP: {zip_names}"
                    )
```

which evaluates to:
```
ValueError: Multiple files found in ZIP file. Only one file per ZIP: ['squirrel_detection.gz', '__MACOSX/._squirrel_detection.gz', 'squirrel_detection.parquet', '__MACOSX/._squirrel_detection.parquet']
```

do you think its better to catch this error and return custom error message from the client?

Reply:
No need to catch it, imo it's fine, looks pretty self explanatory!